app/extensions.js

```diff
@@ -555,6 +592,219 @@ module.exports.init = () => {
   extensionInfo.setState(config.syncExtensionId, extensionStates.REGISTERED)
   loadExtension(config.syncExtensionId, getExtensionsPath('brave'), generateSyncManifest(), 'unpacked')
 
+  if (getSetting(settings.ETHWALLET_ENABLED)) {
+    const gethProcessKey = process.platform === 'win32'
+      ? 'geth.exe'
+      : 'geth'
+    const ipcPath = process.platform === 'win32'
+      ? '\\\\.\\pipe\\geth.ipc'
+      : path.join(app.getPath('userData'), 'ethereum', 'geth.ipc')
+    const pidPath = process.platform === 'win32'
+      ? '\\\\.\\pipe\\geth.pid'
+      : path.join(app.getPath('userData'), 'ethereum', 'geth.pid')
+    const gethProcessPath = path.join(getExtensionsPath('bin'), gethProcessKey)
+
+    const gethDataDir = path.join(app.getPath('userData'), 'ethereum')
+
+    const gethArgs = [
+      '--light',
+      '--rpc',
+      '--ws',
+      '--wsorigins',
+      'chrome-extension://dakeiobolocmlkdebloniehpglcjkgcp',
+      '--datadir',
+      gethDataDir,
+      '--ipcpath',
+      ipcPath
+    ]
+
+    let staticNodes = []
+    if (process.env.ETHEREUM_NETWORK === 'ropsten') {
+      gethArgs.push('--testnet')
+      gethArgs.push('--rpcapi', 'admin,eth,web3')
+      staticNodes.push(
+        'enode://9e0b07fae6615bb0e154591c50f26fddef0833f2a54fa857cc2c20e6b6e2f16fa2ce6b0022309bce73c35110eff97a1684cb92d26a5ac58f4c9378f52e137e1f@52.8.32.174:30303',
+        'enode://b2ca57957bd17dcea46f38648ab71a4c03e0ae4f727c5dbe655d12b7d999a64804125b80963da5e894e6a678ed8eef20d86c89e5737886eec6837b3c6588a8f6@52.8.32.174:30303'
+      )
+    }
+
+    if (!fs.existsSync(gethDataDir)) {
+      fs.mkdirSync(gethDataDir)
+    }
+    if (!fs.existsSync(path.join(gethDataDir, 'geth'))) {
+      fs.mkdirSync(path.join(gethDataDir, 'geth'))
+    }
+    fs.writeFileSync(path.join(gethDataDir, 'geth', 'static-nodes.json'), JSON.stringify(staticNodes))
+
+    const spawnOptions = {
+      stdio: process.env.GETH_LOG ? 'inherit' : 'ignore'
+    }
+
+    var geth
+    let gethProcessId
+    let gethRetryTimeoutId
+    const gethRetryInterval = 30000
+
+    const spawnGeth = () => {
+      // If the process from the previous browswer session still lingers, it should be killed
+      if (fs.existsSync(pidPath)) {
+        try {
+          const pid = fs.readFileSync(pidPath)
+          cleanupGeth(pid)
```

Comment:
since geth is probably undefined at this point, will `cleanupGeth` actually kill that pid?

Reply:
Good catch, in the event that `geth` is undefined the kill will not go forward. Will fix.

Reply:
fixed